deep_water_level/src/deep_water_level/train.py

```diff
@@ -125,25 +126,35 @@ def do_training(
     model.to(device)
     print(f"Model summary: {model}")
 
-    criterion = nn.MSELoss()
-    optimizer = torch.optim.Adam(model.parameters(), lr=learning_rate)
+    optimizer = AdamW(model.parameters(), lr=learning_rate)
 
     for epoch in range(n_epochs):
         # Train for the epoch
         model.train()
+        train_loss = 0
         for i, (inputs, labels, filenames) in enumerate(train_data):
             optimizer.zero_grad()
 
             inputs = inputs.to(device)
             labels = labels.to(device)
             outputs = model(inputs)
-            loss = criterion(outputs, labels)
+            # loss = criterion(outputs, labels)
+            mu, log_var = outputs[:, 0], outputs[:, 1]
+
+            loss = 0.5 * log_var + 0.5 * ((labels[:, 0] - mu) ** 2) / torch.exp(log_var)
```

Comment:
This is the important part

Reply:
Where does the `torch.exp` come from in the denominator of the second part? Shouldn't it be `log_var**2`?

Reply:
oh nevermind.... log_var is already the log of the variance so it `torch.exp` just results in the variance. But shouldn't it be squared?

Reply:
ok ok i see..... log var is "log of variance" is already `log(sigma**2)` so the squared is already in there.